networkd/network_server.ml

```diff
@@ -130,13 +130,25 @@ module Sriov = struct
       else
         Ok Modprobe_successful_requires_reboot
     | None ->
-      debug "%s SR-IOV on a device: %s via sysfs" op dev;
+      (* enable: try sysfs interface to set numfvs = maxvfs. if fails, but vfs are enabled, assume manual configuration.
+         disable: Net.Sriov.disable will not be called for manually configured interfaces, as determined by `require_operation_on_pci_device` *)
+      (* let Ok(maxvfs) = Sysfs.get_sriov_maxvfs dev in *)
+      if enable then
       begin
-        if enable then Sysfs.get_sriov_maxvfs dev
-        else Sysfs.unbind_child_vfs dev >>= fun () -> Ok 0
-      end >>= fun numvfs ->
-      Sysfs.set_sriov_numvfs dev numvfs >>= fun _ ->
-      Ok Sysfs_successful
+        let present_numvfs = Sysfs.get_sriov_numvfs dev in
+        match
+          ( try Sysfs.get_sriov_maxvfs dev >>= fun (maxvfs) -> maxvfs |> Sysfs.set_sriov_numvfs dev
```

Comment:
Minor points:
* `maxvfs` does require parenthesis
* please try to avoid overlong lines

I believe this can be simplified:
```
match Sysfs.get_sriov_maxvfs dev >>= fun (maxvfs) -> maxvfs |> Sysfs.set_sriov_numvfs dev with
| Ok _, _ ->
| Error err, 0 ->
| _, _ ->
| exception e ->
```

Reply:
Has been adapted per your recommendation. `present_numvfs` was missing from the match, but the principle works great. I was not aware of the option to match against exceptions.